hotdoc/extensions/gi/transition_scripts/hotdoc_gtk_doc_porter

```diff
@@ -69,9 +68,9 @@ class Patcher(object):
             return
 
         lines = []
-        with open(filename, 'rb') as _:
+        with open(filename, 'r', encoding='utf8') as _:
             for l in _.readlines():
-                lines.append(unicode_dammit(l.replace(b'\xc3\xa2\xe2\x82\xac\xe2\x84\xa2', b"'")))
+                lines.append(l.replace('\xc3\xa2\xe2\x82\xac\xe2\x84\xa2', "'"))
```

Comment:
If you're reading the file as text, it will never contain `'\xc3\xa2\xe2\x82\xac\xe2\x84\xa2'`. That string is a special sort of byte-encoding damage that happens when you read escaped unicode data and double-decode it to bytes. Here's what happens to a Unicode "\u2019" right single quotation mark, for example:

```python3
>>> import codecs
>>> codecs.escape_decode(codecs.unicode_escape_decode("\u2019")[0])[0]
b'\xc3\xa2\xc2\x80\xc2\x99'
```

I could've written `"’"` instead of `"\u2019"` there and gotten exactly the same answer, because Python3 is unicode-by-default. But with Python2, your strings weren't automatically unicode, so stuff like that could easily happen.

I'm not sure exactly what combination of damage causes the very similar byte string `b'\xc3\xa2\xe2\x82\xac\xe2\x84\xa2'`, but it will never happen when reading in text mode, and anyway the _unicode_ string `'\xc3\xa2\xe2\x82\xac\xe2\x84\xa2'` doesn't even make sense.

Assuming that line is supposed to be replacing (encoding-damaged) U+2019 single quote marks, in Python3 text mode it should just be this:

```suggestion
                lines.append(l.replace("\u2019", "'"))
```

Reply:
Or better yet, perhaps:
```python3
# pre-loop
fix_quotes = str.maketrans({"\u2019": "'"})

# line-by-line
lines.append(l.translate(fix_quotes))
```

Other mappings can be added to the `str.maketrans()` as well, e.g.:
```python3
fix_quotes = str.maketrans({
    "\u2019": "'",
    "\u201c": '"',
    "\u201d": '"',
})
>>> s = "He said, \u201cDon\u2019t.\u201d"
>>> print(s)
He said, “Don’t.”
>>> print(s.translate(fix_quotes))
He said, "Don't."
```

Reply:
Oh, this is much nicer, thank you! Incorporating it.